### PR TITLE
[RLlib] Cleanup examples folder (vol 30): BC pretraining, then PPO finetuning (new API stack with RLModule checkpoints).

### PR DESCRIPTION
**Pull request by sven1977**
Cleanup examples folder (vol 30): BC pretraining, then PPO finetuning (new API stack with RLModule checkpoints).

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/examples/offline_rl/train_w_bc_finetune_w_ppo.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    metric_key = f"{EVALUATION_RESULTS}/{ENV_RUNNER_RESULTS}/{EPISODE_RETURN_MEAN}"
+    stop = {metric_key: 250.0}
+
+    results = run_rllib_example_script_experiment(base_config, args, stop=stop)
+
+    # Extract the RLModule checkpoint.
+    best_result = results.get_best_result(metric_key)
+    rl_module_checkpoint = (
+        Path(best_result.checkpoint.path)
+        / COMPONENT_LEARNER_GROUP
```
Comment (simonsays1980) [inline rllib/examples/offline_rl/train_w_bc_finetune_w_ppo.py:240]: Nice!!!

File: rllib/examples/offline_rl/train_w_bc_finetune_w_ppo.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+| Trial name                 | status     | loc            |   iter |
+|                            |            |                |        |
+|----------------------------+------------+----------------+--------+
+| BC_CartPole-v1_95ba0_00000 | TERMINATED | 127.0.0.1:1515 |     51 |
++----------------------------+------------+----------------+--------+
++------------------+------------------------+------------------------+
+|   total time (s) |    episode_return_mean |   num_env_steps_traine |
+|                  |                        |             d_lifetime |
+|------------------+------------------------|------------------------|
+|          11.4828 |                  250.5 |                  42394 |
```
Comment (simonsays1980) [inline rllib/examples/offline_rl/train_w_bc_finetune_w_ppo.py:47]: Awesome 11 seconds for 250 points. 51 iterations per 11 seconds is pretty fast.

File: rllib/examples/offline_rl/train_w_bc_finetune_w_ppo.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+|                             |            |                |        |
+|-----------------------------+------------+----------------+--------+
+| PPO_CartPole-v1_e07ac_00000 | TERMINATED | 127.0.0.1:6032 |     37 |
++-----------------------------+------------+----------------+--------+
+
++------------------+------------------------+------------------------+
+|   total time (s) |    episode_return_mean |  num_episodes_lifetime |
+|                  |                        |                        |
++------------------+------------------------+------------------------+
+|          32.7647 |                 450.76 |                    406 |
```
Comment (simonsays1980) [inline rllib/examples/offline_rl/train_w_bc_finetune_w_ppo.py:67]: Dom we have by case a number how long it takes to train PPO from zero to 450?

**Review by simonsays1980 — APPROVED** [review]
Awesome example. Maybe adding the avg time for reaching 450 points without pretraining to show the advantage of pretraining.

File: rllib/algorithms/algorithm.py [hunk 4]
```diff
@@ -857,6 +857,7 @@ def setup(self, config: AlgorithmConfig) -> None:
                     env_steps_sampled=self.metrics.peek(
                         NUM_ENV_STEPS_SAMPLED_LIFETIME, default=0
                     ),
+                    rl_module_state=rl_module_state,
```
Comment (sven1977) [inline rllib/algorithms/algorithm.py:860]: This was a bug!

File: rllib/utils/test_utils.py [hunk 5]
```diff
@@ -1362,7 +1362,11 @@ def run_rllib_example_script_experiment(
         args.as_test = True
 
     # Initialize Ray.
-    ray.init(num_cpus=args.num_cpus or None, local_mode=args.local_mode)
+    ray.init(
```
Comment (sven1977) [inline rllib/utils/test_utils.py:1365]: Added reinit error ignore:
In case one calls this utility function twice in an example script.

File: rllib/examples/offline_rl/train_w_bc_finetune_w_ppo.py (hunk 3 above)
Comment (sven1977) [reply rllib/examples/offline_rl/train_w_bc_finetune_w_ppo.py:67]: It's probably not much slower, if at all. But I think this is not the main point here (we know PPO is super fast learning CartPole). The main goal here is to show that:
a) you can use simple custom Models w/o having to sub-class algo-specific RLModule classes!! <- this is huge and thanks to the new RLModule API concept.
b) it doesn't tank (catastrophic forgetting) after transfer from BC to PPO :)
